### ISSUE TITLE
[Plugin] Linguist (Languages)

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#9733]
## Summary

Linguist is the tooling used by GitHub to determine the languages in a repo. Now GitHub and a few other SCM have APIs to pull this data out but others don't. It would be great if we had a plugin to use Linguist directly. 

**Note:** we have built this using Linguist-JS and would like to contribute it. Here is a sample of what this would look like:

![image](https://user-images.githubusercontent.com/67169551/155170086-313a37b3-5d8c-4770-ac35-a519857cefa5.png)

## Project website (if applicable)

Linguist - https://github.com/github/linguist
Linguist-JS - https://github.com/Nixinova/Linguist

## Context

As a user of both Azure DevOps and Bitbucket this data is not available from their APIs. Being able to look at an entity and quickly know what languages it is made of is extremely helpful

**Comment by freben** [comment]
Super nice! Is it looking up that info in realitime for frontend purposes? Sounds like something that might also be useful to have as a driver in the backend, such as a processor that slaps a "java" label in your metadata if linguist clearly shows a java-majority distribution of language use on that particular component's repo.

**Comment by awanlin** [comment]
We built this with a backend that uses the backend-tasks to schedule the processing. It will generate the data for all the entities that has the needed annotation and save it to the database. The backend has an API for the frontend to call to display the data. There is also a POST endpoint that can be used to generate the data, this was handy for testing but also could be used for a possible manual refresh feature at some point.

No reason this can't be adapted for the pattern you suggested though.

We have this working for roughly 600 repos roughly evenly split between Azure DevOps and Bitbucket. The only issue we've run into is large files seem to cause errors, this has been seen in only 2 repos so we just removed the annotation for now
